pkg/ottl/ottlfuncs/func_append.go

```diff
@@ -0,0 +1,136 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type AppendArguments[K any] struct {
+	Target ottl.GetSetter[K]
+	Value  ottl.Optional[ottl.Getter[K]]
+	Values ottl.Optional[[]ottl.Getter[K]]
+}
+
+func NewAppendFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("append", &AppendArguments[K]{}, createAppendFunction[K])
+}
+func createAppendFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*AppendArguments[K])
+	if !ok {
+		return nil, fmt.Errorf("AppendFactory args must be of type *Appendrguments[K]")
+	}
+
+	return Append(args.Target, args.Value, args.Values)
+}
+
+func Append[K any](target ottl.GetSetter[K], value ottl.Optional[ottl.Getter[K]], values ottl.Optional[[]ottl.Getter[K]]) (ottl.ExprFunc[K], error) {
+	if value.IsEmpty() && values.IsEmpty() {
+		return nil, fmt.Errorf("one of optional arguments needs to be provided")
+	}
+
+	return func(ctx context.Context, tCtx K) (any, error) {
+		t, err := target.Get(ctx, tCtx)
+		if err != nil {
+			return nil, err
+		}
+		if t == nil {
+			// target does not exist, init a slice
+			t = make([]string, 0)
```

Comment:
This is an interesting case.  If I write the statement, `append(attributes["test"], [1], [2, 3])` but `test` does not exist`, should the result be creating a new attribute named `test` and appending `[1]`, and `[2, 3]`, but as strings, resulting in `["1", "2", "3"]`?

Would it be better to error if target is not already some kind of slice?  In that case users would have to do

```
- set(attributes["test"], [1])
- append(attributes["test"], [2, 3])
```

Which is more cumbersome, but avoids a typing issue.

If target is nil, could we set the type based on the `value` or `values` type?